onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -1056,7 +1056,52 @@ CUDAExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
       result.push_back(std::make_unique<ComputeCapability>(std::move(sub_graph)));
     }
   }
+
+  if (enable_compiler_) {
+#ifdef USE_TENSORRT
+    auto trt_result = GetTensorrtCapability(graph);
+    if (!trt_result.empty() && !result.empty()) {
+      for (int i = result.size() - 1; i >= 0; i--) {
+        bool in_trt = false;
+        for (auto& capability : trt_result) {
+          for (auto& node : capability->sub_graph->nodes) {
+            if (node == result[i]->sub_graph->nodes[0]) {
+              in_trt = true;
+              break;
+            }
+          }
+        }
+        if (in_trt) {
+          result.erase(result.begin() + i);
```

Comment:
>result.erase(result.begin() + i); [](start = 10, length = 33)

Is this correct? e.g result has [1, 2, 3, 4, 5], remove 2, then remove 3.

Reply:
checked the for loops scope carefully, seems it's correct. But like Ke mentioned as well, it's not easy to read.

---
In reply to: [305946546](https://github.com/microsoft/onnxruntime/pull/1456#discussion_r305946546) [](ancestors = 305946546)